src/index.ts

```diff
@@ -17,27 +20,48 @@ export interface OpenOpts {
   watchForUpdatesHook?: Callback;
 }
 
-const readFile = async (filepath: string): Promise<Buffer> => {
-  return new Promise((resolve, reject) => {
-    const chunks: Buffer[] = [];
```

Comment:
Old implementation used a list of buffers which was driving excessive memory usage + Buffer.concat which apparently is not memory friendly as well.

Now I'm allocating resulting buffer of the size of the file, and filling it with chunks as we go.